godot-core/src/export.rs

```diff
@@ -0,0 +1,139 @@
+    /// ).
+    fn type_string() -> String;
+}
+
+mod export_impls {
+    use super::*;
+    use crate::builtin::*;
+
+    macro_rules! impl_export_by_clone {
+        ($Ty:ty => $variant_type:ident) => {
+            impl ExportDefaultInfo for $Ty {}
+
+            impl Export for $Ty {
+                fn export(&self) -> Self {
+                    // If `Self` does not implement `Clone`, this gives a clearer error message
+                    // than simply `self.clone()`.
+                    Clone::clone(self)
+                }
+            }
+
+            impl TypeString for $Ty {
+                fn type_string() -> String {
+                    format!("{}:", sys::VariantType::$variant_type as i32)
+                }
+            }
+        };
+    }
+
+    impl_export_by_clone!(Aabb => Aabb);
+    impl_export_by_clone!(bool => Bool);
+    impl_export_by_clone!(Basis => Basis);
+    impl_export_by_clone!(Vector2 => Vector2);
+    impl_export_by_clone!(Vector3 => Vector3);
+    impl_export_by_clone!(Vector4 => Vector4);
+    impl_export_by_clone!(Vector2i => Vector2i);
+    impl_export_by_clone!(Vector3i => Vector3i);
+    impl_export_by_clone!(Quaternion => Quaternion);
+    impl_export_by_clone!(Color => Color);
+    impl_export_by_clone!(GodotString => String);
+    impl_export_by_clone!(StringName => StringName);
+    impl_export_by_clone!(NodePath => NodePath);
+    impl_export_by_clone!(PackedByteArray => PackedByteArray);
+    impl_export_by_clone!(PackedInt32Array => PackedInt32Array);
+    impl_export_by_clone!(PackedInt64Array => PackedInt64Array);
+    impl_export_by_clone!(PackedFloat32Array => PackedFloat32Array);
+    impl_export_by_clone!(PackedFloat64Array => PackedFloat64Array);
+    impl_export_by_clone!(PackedStringArray => PackedStringArray);
+    impl_export_by_clone!(PackedVector2Array => PackedVector2Array);
+    impl_export_by_clone!(PackedVector3Array => PackedVector3Array);
+    impl_export_by_clone!(PackedColorArray => PackedColorArray);
+    impl_export_by_clone!(Plane => Plane);
+    impl_export_by_clone!(Projection => Projection);
+    impl_export_by_clone!(Rid => Rid);
+    impl_export_by_clone!(Rect2 => Rect2);
+    impl_export_by_clone!(Rect2i => Rect2i);
+    impl_export_by_clone!(Transform2D => Transform2D);
+    impl_export_by_clone!(Transform3D => Transform3D);
+    impl_export_by_clone!(f64 => Float);
+    impl_export_by_clone!(f32 => Float);
+    impl_export_by_clone!(i64 => Int);
+    impl_export_by_clone!(i32 => Int);
+    impl_export_by_clone!(i16 => Int);
+    impl_export_by_clone!(i8 => Int);
+    impl_export_by_clone!(u32 => Int);
+    impl_export_by_clone!(u16 => Int);
+    impl_export_by_clone!(u8 => Int);
+
+    // Callables can be exported, however you can't do anything with them in the editor.
+    // But we do need to be able to export them since we can't make something a property without exporting.
```

Comment:
> since we can't make something a property without exporting.

Is that still true, after we discovered we can hide them from the editor via flag?

Reply:
it isnt strictly true no, but until we change this it's true. since we currently do need anything that's exposed as a property to implemented Export.

Reply:
Could you add a todo then, otherwise it is forgotten for sure.

(Not that we actively track TODOs, but we track comments even less 😉)

Reply:
This one would still be nice 🙂